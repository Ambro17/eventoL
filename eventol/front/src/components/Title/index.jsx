import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';


const Title = ({label, children}) => (
  <div className='row'>
    <div className='col-sm-12'>
      <div className='title'>
        <h1>{label}</h1>
        <div className='buttons pull-right'>
          {children}
        </div>
      </div>
    </div>
  </div>
);

Title.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.element,
  ]).isRequired, // TODO: search children default proptypes
  label: PropTypes.string.isRequired,
};
