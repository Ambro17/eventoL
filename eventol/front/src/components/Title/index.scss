.buttons {
  margin-top: -60px;
}

.title {
  text-align: left;
}
