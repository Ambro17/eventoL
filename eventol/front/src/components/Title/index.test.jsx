import React from 'react';
import renderer from 'react-test-renderer';

import Title from '.';

describe('Title', () => {
  let component;
  let instance;
  let tree;

  const getComponent = () => {
    component = renderer.create(<Title />);
    return component;
  };

  beforeEach(() => {
    component = getComponent();
    instance = renderer.root;
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
