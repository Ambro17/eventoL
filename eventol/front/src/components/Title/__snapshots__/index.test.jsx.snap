// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Default Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <div
      className="title"
    >
      <h1>
        
      </h1>
      <div
        className="buttons pull-right"
      />
    </div>
  </div>
</div>
`;
